pkg/rc/replication_controller.go

```diff
@@ -256,6 +262,17 @@ func (rc *replicationController) addPods(current types.PodLocations, eligible []
 			txn, cancelFunc = rc.newAuditingTransaction(context.Background(), txn.Nodes())
 		}
 		if len(possibleSorted) < i+1 {
+			if rc.AllocationStrategy == CattleStrategy {
```

Comment:
it's probably fine to just leave this out of the PR, you can keep it in a branch for later